### PR TITLE
Support for mac with ARM processors

### PR DESCRIPTION
**Pull request by IAvecilla**
This pull request introduces support for Mac computers with M1/M2 processors.

### CODE REVIEW

File: plugins/lsp_rust.lua [hunk 1]
```diff
@@ -9,7 +9,11 @@ local filename
 if PLATFORM == "Windows" then
   filename = "rust-analyzer.exe"
 elseif PLATFORM == "Mac OS X" then
-  filename = "rust-analyzer-x86_64-apple-darwin"
+  if ARCH == "aarch64-darwin" then
+    filename = "rust-analyzer-aarch64-apple-darwin"
+  else
```
Comment (Guldoman) [inline plugins/lsp_rust.lua:14]: Nit: whitespace at the end.

File: manifest.json [hunk 2]
```diff
@@ -10,6 +10,11 @@
         "arch": "x86_64-darwin",
         "checksum": "fdbf094e07fb2133d7ee58f2bacdca2dd7b112bcf9b9c078abb496f845f72f5a"
       },
+      {
+        "url": "https://github.com/LuaLS/lua-language-server/releases/download/3.6.25/lua-language-server-3.6.25-darwin-arm64.tar.gz",
+        "arch": "aarch64-darwin",
+        "checksum": "SKIP"
```
Comment (Guldoman) [inline manifest.json:16]: You can try with `shasum -a 256 filename` or `openssl sha256 filename`.

Comment (IAvecilla) [reply manifest.json]: Nice! Thank you

**Comment by Guldoman** [comment]
Looks good! Thank you!

**Pull request merged**